.github/workflows/test-template.yml

```diff
@@ -0,0 +1,1368 @@
+          name: generator-helper-${{ matrix.os }}
+
+      - run: pnpm run test
+        name: 'get-platform'
+        working-directory: packages/get-platform
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - uses: codecov/codecov-action@v3
+        with:
+          files: ./packages/get-platform/src/__tests__/coverage/clover.xml
+          flags: get-platform,${{ matrix.os }},library,binary
+          name: get-platform-${{ matrix.os }}
+
+      - run: pnpm run test
+        name: 'fetch-engine'
+        working-directory: packages/fetch-engine
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - uses: codecov/codecov-action@v3
+        with:
+          files: ./packages/fetch-engine/src/__tests__/coverage/clover.xml
+          flags: fetch-engine,${{ matrix.os }},library,binary
+          name: fetch-engine-${{ matrix.os }}
+
+      - run: pnpm run test
+        name: 'engines'
+        working-directory: packages/engines
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - uses: codecov/codecov-action@v3
+        with:
+          files: ./packages/engines/src/__tests__/coverage/clover.xml
+          flags: engines,${{ matrix.os }},library,binary
+          name: engines-${{ matrix.os }}
+
+      - run: pnpm run test
+        name: 'instrumentation'
+        working-directory: packages/instrumentation
+        env:
+          CI: true
+          SKIP_GIT: true
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
+      - uses: codecov/codecov-action@v3
+        with:
+          files: ./packages/instrumentation/src/__tests__/coverage/clover.xml
+          flags: instrumentation,${{ matrix.os }},library,binary
+          name: instrumentation-${{ matrix.os }}
+
+  #
+  # Run all tests on macOS and Windows.
+  #
+  # Unlike the other jobs, this job doesn't use Docker (and thus skips some
+  # tests that require dependencies not easily installable without Docker).
+  #
+  # It also runs most tests for different packages sequentially
+  # to prevent the combinatorial explosion of the number of parallel jobs,
+  # to minimize the number of times we need to install MySQL using the package manager
+  # (PostgreSQL and MongoDB are provided by GitHub Actions out of the box), and
+  # minimize the time spent waiting for a free runner.
+  #
+  no-docker-client-functional:
+    timeout-minutes: 40
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest]
+        node: [14]
+        queryEngine: ${{ fromJson(needs.determineBuildMatrixParams.outputs.queryEngine) }}
+        # Commented out as we still have limited capacity for macOS runners
+        # When we have more capacity we can shard again
+        # shard: ['1', '2']
```

Comment:
Maybe we can shard these now as we should have capacity?
(can be done separately)

Reply:
Yes, i'd rather do this separately